### ISSUE TITLE
Check if we regressed in startup performance

### ISSUE THREAD

**Issue by bpasero** [Microsoft/vscode#64195]
Given the results from our Windows perf machine.

/cc @jrieken

**Comment by jrieken** [comment]
Startup telemetry doesn't show anything yet, best would be to drill into the perf canary numbers. I can let you know how...

**Comment by bpasero** [comment]
@jrieken I can do that but I thought about setting up my windows box to run these tests and compare the numbers. However for that it would be useful to know the exact configuration and properties that make this launch comparable.

**Comment by jrieken** [comment]
It's a standard startup: explorer visible, one file open, no panels, no extensions, cached data. The perf canary uses the 'prof-append-timers' flag

**Comment by bpasero** [comment]
@jrieken is there something in the settings file? I remember you mentioned a setting to prevent the panel from opening

**Comment by jrieken** [comment]
> I remember you mentioned a setting to prevent the panel from opening

No, it's a hand-tuned state in a special user data dir...

**Comment by bpasero** [comment]
I ran some tests on my Windows laptop to compare stable and insiders and could not find a slowdown that popped up into my face (tried both cold startup as well as window reload). In fact, workbench ready was reported sooner (in both cases against a fresh user data dir, without extensions having one file open).

Also it also looks like our performance bot is pretty much back to normal:

![image](https://user-images.githubusercontent.com/900690/49527960-47aaa400-f8b3-11e8-967a-6162fcd35a56.png)

Since I collect performance numbers around the changes to local storage via telemetry, I will have some data to look at once we ship stable. One performance optimization I did not yet look at is to move the SQLite storage (for workspaces) out of the renderer into the main process to load it in parallel to the window loading. The downside is that it would put more pressure on the main process. But I am open to ideas here how to improve that.

In the startup performance metrics I added more data:
* global storage init through our SQLite (in main)
* workspace storage init through our SQLite (in renderer - happens in parallel to configuration service loading)

![image](https://user-images.githubusercontent.com/900690/49528299-ffd84c80-f8b3-11e8-9d30-fdd047e517e1.png)

I think the real benefit of this work only kicks in once we ship with Electron 3 where `window.localStorage` is reset to be empty and as such no longer loaded on startup.
